diff --git a/PVE/QemuServer.pm b/PVE/QemuServer.pm
index 1c0fba2..0332213 100644
--- a/PVE/QemuServer.pm
+++ b/PVE/QemuServer.pm
@@ -1179,6 +1179,12 @@ EODESCR
 	optional => 1,
 	default => 0,
     },
+    'use-root-port' => {
+       type => 'boolean',
+       description => "Plug PCI-express device directly (without PCI bridge).",
+       optional => 1,
+       default => 0,
+    },
 };
 PVE::JSONSchema::register_format('pve-qm-hostpci', $hostpci_fmt);
 
@@ -3257,7 +3263,7 @@ sub config_to_command {
 	my $pcie = $d->{pcie};
 	if($pcie){
 	    die "q35 machine model is not enabled" if !$q35;
-	    $pciaddr = print_pcie_addr("hostpci$i");
+	    $pciaddr = print_pcie_addr("hostpci$i", $d->{'use-root-port'});
 	}else{
 	    $pciaddr = print_pci_addr("hostpci$i", $bridges);
 	}
diff --git a/PVE/QemuServer/PCI.pm b/PVE/QemuServer/PCI.pm
index 0d9cf24..6972d03 100644
--- a/PVE/QemuServer/PCI.pm
+++ b/PVE/QemuServer/PCI.pm
@@ -126,15 +126,26 @@ sub print_pci_addr {
 }
 
 sub print_pcie_addr {
-    my ($id) = @_;
+    my ($id, $no_bridge) = @_;
 
     my $res = '';
-    my $devices = {
-	hostpci0 => { bus => "ich9-pcie-port-1", addr => 0 },
-	hostpci1 => { bus => "ich9-pcie-port-2", addr => 0 },
-	hostpci2 => { bus => "ich9-pcie-port-3", addr => 0 },
-	hostpci3 => { bus => "ich9-pcie-port-4", addr => 0 },
-    };
+    my $devices;
+
+    if ($no_bridge) {
+	$devices = {
+	    hostpci0 => { bus => "pcie.0", addr => 16 },
+	    hostpci1 => { bus => "pcie.0", addr => 17 },
+	    hostpci2 => { bus => "pcie.0", addr => 18 },
+	    hostpci3 => { bus => "pcie.0", addr => 19 },
+	};
+    }else{
+	$devices = {
+	    hostpci0 => { bus => "ich9-pcie-port-1", addr => 0 },
+	    hostpci1 => { bus => "ich9-pcie-port-2", addr => 0 },
+	    hostpci2 => { bus => "ich9-pcie-port-3", addr => 0 },
+	    hostpci3 => { bus => "ich9-pcie-port-4", addr => 0 },
+	};
+    }
 
     if (defined($devices->{$id}->{bus}) && defined($devices->{$id}->{addr})) {
 	   my $addr = sprintf("0x%x", $devices->{$id}->{addr});
